state/upgrades.go

```diff
@@ -1430,3 +1430,45 @@ func CopyMongoSpaceToHASpaceConfig(st *State) error {
 	_, err = settings.Write()
 	return errors.Annotate(err, "writing controller info")
 }
+
+// CreateMissingApplicationConfig ensures that all models have an application config in the db.
+func CreateMissingApplicationConfig(st *State) error {
+	settingsColl, settingsCloser := st.db().GetRawCollection(settingsC)
+	defer settingsCloser()
+
+	var applicationConfigIDs []struct {
```

Comment:
This is actually any Object ID, and it seems a bit confusing to read all settings ids and treat them "applicationConfigIDs". IIRC, settings holds lots of things like model config, etc.
Maybe just treat this as
var allSettingIDs?

Reply:
I've fixed the find to only grab IDs that are for application configs.